sql/sp_head.cc

```diff
@@ -1345,6 +1345,65 @@ sp_head::execute(THD *thd, bool merge_da_on_success)
 #endif /* WITH_WSREP */
     err_status= i->execute(thd, &ip);
 
+#ifdef WITH_WSREP
+    if (WSREP(thd))
+    {
+      if (((thd->wsrep_trx().state() == wsrep::transaction::s_executing) &&
```

Comment:
This part is questionable. I understand that we need to return error to upper layer in case of ::s_executing, but what if we are in ::s_must_abort but thd is fatal error or killed ? There could be more precise comment what these differnt cases here actually mean. Secondly, why ::s_must_replay is here as I though it is handled on else ?